x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/queries/SemanticSparseVectorQueryRewriteInterceptor.java

```diff
@@ -0,0 +1,99 @@
+
+    public static final NodeFeature SEMANTIC_SPARSE_VECTOR_QUERY_REWRITE_INTERCEPTION_SUPPORTED = new NodeFeature(
+        "search.semantic_sparse_vector_query_rewrite_interception_supported"
+    );
+
+    public SemanticSparseVectorQueryRewriteInterceptor() {}
+
+    @Override
+    public QueryBuilder interceptAndRewrite(QueryRewriteContext context, QueryBuilder queryBuilder) {
+        assert (queryBuilder instanceof SparseVectorQueryBuilder);
+        SparseVectorQueryBuilder sparseVectorQueryBuilder = (SparseVectorQueryBuilder) queryBuilder;
+        QueryBuilder rewritten = queryBuilder;
+        InferenceIndexInformationForField inferenceIndexInformationForField = SemanticQueryInterceptionUtils.resolveIndicesForField(
+            sparseVectorQueryBuilder.getFieldName(),
+            context.getResolvedIndices()
+        );
+
+        if (inferenceIndexInformationForField == null || inferenceIndexInformationForField.inferenceIndices().isEmpty()) {
+            // No inference fields, return original query
+            return rewritten;
+        } else if (inferenceIndexInformationForField.nonInferenceIndices().isEmpty() == false) {
+            // Combined inference and non inference fields
+            BoolQueryBuilder boolQueryBuilder = new BoolQueryBuilder();
+            boolQueryBuilder.should(
+                SemanticQueryInterceptionUtils.createSubQueryForIndices(
+                    inferenceIndexInformationForField.nonInferenceIndices(),
+                    SemanticQueryInterceptionUtils.createSubQueryForIndices(
+                        inferenceIndexInformationForField.nonInferenceIndices(),
+                        sparseVectorQueryBuilder
+                    )
+                )
+            );
+            // We always perform nested subqueries on semantic_text fields, to support
+            // sparse_vector queries using query vectors
+            boolQueryBuilder.should(
+                SemanticQueryInterceptionUtils.createSubQueryForIndices(
+                    inferenceIndexInformationForField.inferenceIndices(),
+                    buildNestedQueryFromSparseVectorQuery(sparseVectorQueryBuilder)
+                )
+            );
+            rewritten = boolQueryBuilder;
+        } else {
+            // Only semantic text fields
+            rewritten = buildNestedQueryFromSparseVectorQuery(sparseVectorQueryBuilder);
+        }
+
+        return rewritten;
+    }
+
+    private QueryBuilder buildNestedQueryFromSparseVectorQuery(SparseVectorQueryBuilder sparseVectorQueryBuilder) {
+        return QueryBuilders.nestedQuery(
+            getNestedFieldPath(sparseVectorQueryBuilder.getFieldName()),
+            new SparseVectorQueryBuilder(
+                getNestedEmbeddingsField(sparseVectorQueryBuilder.getFieldName()),
+                sparseVectorQueryBuilder.getQueryVectors(),
+                sparseVectorQueryBuilder.getInferenceId(),
```

Comment:
This should be retrieved from the index metadata, as per Ben's comment this is not necessary